plugins/jobs/web_client/views/JobListWidget.js

```diff
@@ -90,99 +157,246 @@ var JobListWidget = View.extend({
+        if (this.currentView === 'phase' || this.currentView === 'time') {
+            this.$('.g-main-content').html(JobsGraphWidgetTemplate());
+            this.phaseFilterWidget.setValues(this.phasesFilter);
+            this.phaseFilterWidget.setElement(this.$('.graph-filter-container .phase')).render();
         }
 
+        this._renderData();
+
         return this;
     },
 
-    _statusChange: function (event) {
-        var job = event.data,
-            tr = this.$('tr[g-job-id=' + job._id + ']');
+    _renderData: function () {
+        var jobs = this.collection.toArray();
 
-        if (!tr.length) {
+        if (!jobs.length) {
+            this.$('.g-main-content,.g-job-pagination').hide();
+            this.$('.g-no-record').show();
             return;
+        } else {
+            this.$('.g-main-content,.g-job-pagination').show();
+            this.$('.g-no-record').hide();
         }
 
-        if (this.columns & this.columnEnum.COLUMN_STATUS_ICON) {
-            tr.find('td.g-status-icon-container').attr('status', job.status)
-              .find('i').removeClass().addClass(JobStatus.icon(job.status));
+        if (this.currentView === 'list') {
+            this.$('.g-main-content').html(JobListWidgetTemplate({
+                jobs: jobs,
+                showHeader: this.showHeader,
+                columns: this.columns,
+                columnEnum: this.columnEnum,
+                linkToJob: this.linkToJob,
+                triggerJobClick: this.triggerJobClick,
+                JobStatus: JobStatus,
+                formatDate: formatDate,
+                DATE_SECOND: DATE_SECOND
+            }));
         }
-        if (this.columns & this.columnEnum.COLUMN_STATUS) {
-            tr.find('td.g-job-status-cell').text(JobStatus.text(job.status));
-        }
-        if (this.columns & this.columnEnum.COLUMN_UPDATED) {
-            tr.find('td.g-job-updated-cell').text(
-                formatDate(job.updated, DATE_SECOND));
+
+        var changeScaleType = view => {
+            return (event, item) => {
+                if (item && item.itemName === 'ylabel') {
+                    view.destroy(); this.yScale = this.yScale === 'sqrt' ? 'linear' : 'sqrt'; this.render();
+                }
+            };
+        };
+
+        var openDetailView = view => {
+            return (event, item) => {
+                if (item && (item.itemName === 'bar' || item.itemName === 'circle')) {
+                    window.open(`#/job/${item.datum.id}`, '_blank');
+                }
+            };
+        };
+
+        if (this.currentView === 'phase') {
+            new JobStatusSegmentizer().segmentize(jobs);
+            let vegaData = this._prepareDataForChart(jobs);
+
+            let config = jQuery.extend(true, {}, phaseChartConfig);
+            config.width = Math.min(Math.max(this.$el.width() - 50, jobs.length * 16 + 100), jobs.length * 30 + 400);
```

Comment:
where do these numbers come from?  @zachmullen is there a better way to do this?